src/ekat/util/ekat_type_traits.hpp

```diff
@@ -24,8 +26,8 @@ namespace ekat {
  * especially useful for bfb tests agaisnt fortran,
  * to ensure that literals are not a source of round-off differences.
  */
-template<typename T> KOKKOS_INLINE_FUNCTION
-constexpr typename std::enable_if<std::is_arithmetic<T>::value,Real>::type
+template<typename T, bool doublePrecision> KOKKOS_INLINE_FUNCTION
```

Comment:
We're not considering half-precision floating point numbers, are we? If we are, using a bool to determine whether something is double precision or not might not give enough info.

Reply:
I thought about this. I even started writing an enum for the precision. However, I refrained myself, since I wasn't sure supporting half precision was ever going to be useful in the E3SM ecosystem.

That said, supporting it with an enum would not be that hard. But if we show it in the enum, we should make sure all our utils and algorithms either support it, or throw errors (possibly at compile time) if one tries to use them with half precision.

Reply:
Half precision has been discussed on the EAGLES project, but it's still just an idea. I didn't know if other E3SM subprojects were toying with it too. What do you think? Should we wait to hear that someone needs it and change in a later PR if/when we decide to, maybe?

Reply:
I'm definitely interested in half precision but think we should just focus on "get something working" now. Thus if it is easy to generalize things now to make half precision easier later, we should do it. Otherwise, we should save these things for a future PR.

Reply:
Let's open an issue in Ekat, to keep track of the idea (and the needs for that idea). But for now, I'm also for getting stuff going.

But please, feel free to formally request it on behalf of EAGLES (or whatever project), if you move forward with the idea.

Reply:
Opened a feature request.